### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by lakshit07** [dasebe/webcachesim#8]
g++  -std=c++11  -MMD -MP  -I./ -Wall -Werror  -O2  -c -o caches/lru_variants.o caches/lru_variants.cpp
In file included from caches/lru_variants.cpp:5:
In file included from caches/lru_variants.h:6:
./cache.h:79:26: error: moving a temporary object prevents copy elision
      [-Werror,-Wpessimizing-move]
        Cache_instance = move(get_factory_instance()[name]->create_unique());
                         ^
./cache.h:79:26: note: remove std::move call here
        Cache_instance = move(get_factory_instance()[name]->create_unique());
                         ^~~~~                                             ~
1 error generated.
make: *** [caches/lru_variants.o] Error 1

**Comment by dasebe** [comment]
Hi @lakshit07,
which compiler are you using?
I just checked the current version and with g++ (GCC) 7.2.0 there are no compilation warnings or errors. I've also checked GCC 6, and all work as well.

Are you maybe using LLVM/Clang? I don't typically use that compiler, but I'm happy to help debug and fix. Does Clang's suggestion work for you?

**Comment by lakshit07** [comment]
Hi @dasebe 
On running `g++ --version`  I get 
> Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr  /include/c++/4.2.1
> Apple LLVM version 9.0.0 (clang-900.0.39.2)
> Target: x86_64-apple-darwin17.4.0
> Thread model: posix
> InstalledDir: /Library/Developer/CommandLineTools/usr/bin

Which Clang suggestion are you referring to?

**Comment by tnsardesai** [comment]
Hi @dasebe, I encountered the same issue when trying to compile using both g++ and clang++ on OSX. I found the root cause is the move statements as mentioned in the error by OP. The reason you aren't seeing it in GCC 7.2.0 is because `-Wall` doesn't turn on `-Wpessimizing-move` which catches these kinds of warnings. The latest version of GCC, clang (the default versions on OSX) turns on `-Wpessimizing-move` in `-Wall`.

Reference docs:
[GCC 7.2.0](https://gcc.gnu.org/onlinedocs/gcc-7.2.0/gcc/Invoking-GCC.html#Invoking-GCC)
[GCC 10.0.0](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html)

Fix is to remove the call to move in `cache.h` at line 79 and `webcachesim.cpp` at line 24.
